### PR TITLE
🏗 Assorted storybook fixes

### PR DESCRIPTION
**Pull request by rsimha**
This is what the storybook build [currently looks like](https://app.circleci.com/pipelines/github/rsimha/amphtml/168/workflows/c48a4c90-6006-479e-b8e2-0cd247d547c0/jobs/586/parallel-runs/0/steps/0-107) during CI:

image

**A few fixes:** (in anticipation of the move to CircleCI)
- Makes `build-storybook` synchronous so that build steps and the gulp task do not early exit before completion
- Leaves `start-storybook` as asynchronous, but updates task invocation code and cleans up flags
- Removes the explicit `--ci` flag for `start-storybook` and auto-detects CI runs
- Adds an implicit `--loglevel` flag to `build-storybook` to silence verbose logging during CI
- Modernizes webpack configs to eliminate deprecation warnings
- Includes `examples/storybook` in `gulp clean`
- Fixes a bug in the default `gulp` lazy-builder so all extensions aren't unnecessarily pre-built (speeds up storytime launch)
- Changes `storybook/package.json` to have `devDependencies` in order to enable renovate (most packages are outdated)

**New look (CI):**

image

**New look (local):**

image

**References:**
- https://storybook.js.org/docs/react/api/cli-options
- https://storybook.js.org/docs/react/configure/webpack#extending-storybooks-webpack-config

### CODE REVIEW

File: build-system/server/lazy-build.js [hunk 1]
```diff
@@ -139,7 +139,7 @@ async function preBuildRuntimeFiles() {
  * Pre-builds default extensions and ones requested via command line flags.
  */
 async function preBuildExtensions() {
-  const extensions = getExtensionsToBuild();
+  const extensions = getExtensionsToBuild(/* preBuild */ true);
```
Comment (alanorozco) [inline build-system/server/lazy-build.js:142]: Is this related to this change?

**Review by alanorozco — APPROVED** [review]

File: build-system/server/lazy-build.js (hunk 1 above)
Comment (rsimha) [reply build-system/server/lazy-build.js:142]: This fixes a bug in the lazy builder that affects storybook. (Launching storybook would cause all ~150 extensions to be built, but now we quickly start up the `gulp` server and build extensions on-demand.)
